src/prefect/client/orchestration/__init__.py

```diff
@@ -1182,8 +1183,17 @@ async def raise_for_api_version_mismatch(self) -> None:
         if api_version.major != client_version.major:
             raise RuntimeError(
                 f"Found incompatible versions: client: {client_version}, server: {api_version}. "
-                f"Major versions must match."
+                "Major versions must match."
             )
+        if api_version < client_version:
```

Comment:
would we want to suppress this for `dirty` versions?

Reply:
Maybe? Only people actively working on `prefect` would see the warning for non-released versions. We can add handling for that if it's annoying.

Reply:
Yea I vote we keep the comparison simple for now and change if it gets too noisy